gulp/tasks/html.js

```diff
@@ -1,21 +1,36 @@
 const fs = require('fs');
+
+// stringify functions https://gist.github.com/cowboy/3749767
+var stringify = function (obj, prop) {
+  var placeholder = '____PLACEHOLDER____';
+  var fns = [];
+  var json = JSON.stringify(obj, function (key, value) {
+    if (typeof value === 'function') {
+      fns.push(value);
+      return placeholder;
+    }
+    return value;
+  }, 2);
+  json = json.replace(new RegExp('"' + placeholder + '"', 'g'), function () {
+    return fns.shift();
+  });
+  return 'this["' + prop + '"] = ' + json + ';';
+};
+
 module.exports = function (gulp, plugins, config, env) {
   return function html() {
     return gulp.src(env.production() ? config.build + '/*.html' : 'html/*.html')
-      .pipe(plugins.inject(gulp.src(['config.json']), {
+      .pipe(plugins.inject(gulp.src(['config.js']), {
         starttag: '<!-- inject:config -->',
-        transform: function (filePath, customConfig) {
-          var defaultConfig = fs.readFileSync('config.default.json', 'utf8');
-          var buildConfig = Object.assign(
-            JSON.parse(JSON.minify(defaultConfig)),
-            JSON.parse(JSON.minify(customConfig.contents.toString('utf8')))
-          );
+        transform: function () {
+          delete require.cache[require.resolve('../../config.default')];
+          delete require.cache[require.resolve('../../config')];
+          var buildConfig = Object.assign({}, require('../../config.default')(), require('../../config')());
```

Comment:
kann man nicht `{},` wegwerfen? die `config.default.js` sollte es doch immer geben.

Reply:
Both are not variables and need to be assigned anyway. First argument will be assigned the merged object.